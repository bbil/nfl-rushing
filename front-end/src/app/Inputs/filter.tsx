import React from 'react';
import { connect } from 'react-redux';
import { applyFilter, clearFilter } from '../../redux/rushing/thunk';

export interface FilterProps {
    setFilter: (filter: string) => void;
    clearFilter: () => void;
}

export const FilterComponent: React.FC<FilterProps> = props => {
    const [nameFilter, setNameFilter] = React.useState<string>('');
    const onNameFilterChange = React.useCallback(
        (event: React.ChangeEvent<HTMLInputElement>) => {
            setNameFilter(event.target.value);
    }, [setNameFilter]);

    const clearFilter = () => {
        props.clearFilter();
        setNameFilter('');
    };

    return (
        <div>
            <h3>Player Name Filter:</h3>
            <input id='filter-input' value={nameFilter} onChange={onNameFilterChange} />
            <button id='filter-ok' disabled={!nameFilter} onClick={() => props.setFilter(nameFilter)}>OK</button>
            <button id='filter-clear' onClick={clearFilter}>Clear Filter</button>
        </div>
    );
};

function mapDispatchToProps(dispatch: Function) {
    return {
        setFilter: (filter: string) => dispatch(applyFilter(filter)),
        clearFilter: () => dispatch(clearFilter())
    }
}

export const Filter = connect(undefined, mapDispatchToProps)(FilterComponent);