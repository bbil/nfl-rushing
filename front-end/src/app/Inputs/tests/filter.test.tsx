import React from 'react';
import '@testing-library/react';

import { shallow } from 'enzyme';

import { FilterProps, FilterComponent } from '../filter';

describe('Filter Component', () => {
    const PROPS: FilterProps = {
        clearFilter: jest.fn(),
        setFilter: jest.fn()
    };

    afterEach(() => {
        jest.resetAllMocks();
    });

    it('renders default; no value, and button disabled', () => {
        const wrapper = shallow(<FilterComponent {...PROPS} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('enables button after input', () => {
        const wrapper = shallow(<FilterComponent {...PROPS} />);

        wrapper.find('#filter-input').simulate('change', { target: { value: 'new value' }});

        expect(wrapper).toMatchSnapshot();
    });

    it('calls set filter on button click', () => {
        const wrapper = shallow(<FilterComponent {...PROPS} />);

        wrapper.find('#filter-input').simulate('change', { target: { value: 'new value' }});
        wrapper.find('#filter-ok').simulate('click');

        expect(PROPS.setFilter).toHaveBeenCalledTimes(1);
        expect(PROPS.setFilter).toHaveBeenCalledWith('new value');
    });

    it('calls clear filter on button click, and sets value back to empty string', () => {
        const wrapper = shallow(<FilterComponent {...PROPS} />);

        wrapper.find('#filter-input').simulate('change', { target: { value: 'new value' }});
        wrapper.find('#filter-clear').simulate('click');

        expect(PROPS.clearFilter).toHaveBeenCalledTimes(1);

        expect(wrapper).toMatchSnapshot();
    });
});

