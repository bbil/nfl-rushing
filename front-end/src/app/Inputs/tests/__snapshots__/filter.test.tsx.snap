// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component calls clear filter on button click, and sets value back to empty string 1`] = `
<div>
  <h3>
    Player Name Filter:
  </h3>
  <input
    id="filter-input"
    onChange={[Function]}
    value=""
  />
  <button
    disabled={true}
    id="filter-ok"
    onClick={[Function]}
  >
    OK
  </button>
  <button
    id="filter-clear"
    onClick={[Function]}
  >
    Clear Filter
  </button>
</div>
`;

exports[`Filter Component enables button after input 1`] = `
<div>
  <h3>
    Player Name Filter:
  </h3>
  <input
    id="filter-input"
    onChange={[Function]}
    value="new value"
  />
  <button
    disabled={false}
    id="filter-ok"
    onClick={[Function]}
  >
    OK
  </button>
  <button
    id="filter-clear"
    onClick={[Function]}
  >
    Clear Filter
  </button>
</div>
`;

exports[`Filter Component renders default; no value, and button disabled 1`] = `
<div>
  <h3>
    Player Name Filter:
  </h3>
  <input
    id="filter-input"
    onChange={[Function]}
    value=""
  />
  <button
    disabled={true}
    id="filter-ok"
    onClick={[Function]}
  >
    OK
  </button>
  <button
    id="filter-clear"
    onClick={[Function]}
  >
    Clear Filter
  </button>
</div>
`;
